scripts/util_functions.sh

```diff
@@ -418,6 +418,14 @@ get_flags() {
     fi
   fi
   [ -z $RECOVERYMODE ] && RECOVERYMODE=false
+  if [ -z $DYNAMIC_PARTITIONS ]; then
+    DYNAMIC_PARTITIONS=$(getprop ro.boot.dynamic_partitions)
```

Comment:
Don't getprop in twrp, because twrp may be made based on an old system(before retrofitted dynamic partitions).

Reply:
I previously used `$(grep_get_prop ro.boot.dynamic_partitions /default.prop /vendor/build.prop)` however this wil probably not work either if TWRP was built without dynamic partitions, as it won't be able to mount the partitions.

What do you suggest to do instead?

Reply:
#7147

Reply:
Unfortunately, we don't have any means. The retrofitted dynamic partition device does not have a super block, and the twrp built on old system cannot mount system partition and has no ro.boot.dynamic_partitions prop. Fairphone's bootloader does not pass the correct parameter, it depends on the kernel, but twrp may use an old kernel.

Reply:
I have been working on updating FP3 TWRP to support ROMS with and without dynamic partitions.
There I am checking for "androidboot.super_partition" in the boot.img that is installed in the current slot.
Maybe that would be an option?

Something along these lines:

```
BOOTDEVICE=/dev/block/bootdevice/by-name/boot$(getprop ro.boot.slot_suffix)
dd if="$BOOTDEVICE" bs=2048 count=1 | grep -q androidboot.super_partition
if [ $? -ne 0 ]; then
    echo "ROM does not use dynamic partitions"
else
    echo "ROM uses dynamic partitions"
fi
```